src/main/java/com/yelp/nrtsearch/server/search/collectors/DocCollector.java

```diff
@@ -67,8 +67,11 @@ public DocCollector(
   }
 
   public static int computeNumHitsToCollect(SearchRequest request) {
-    // determine how many hits to collect based on request, facets and rescore window
+    // determine how many hits to collect based on request, facets, rescore window and hits to log
     int collectHits = request.getTopHits();
+    if (request.hasLoggingHits()) {
+      collectHits = request.getLoggingHits().getHitsToLog();
+    }
```

Comment:
What if `hitsToLog` is less than `topHits`? This might be better as a `max` operation

Reply:
++ I missed adding `Max` operation here